### DISCUSSION TITLE
Variant not showing

### DISCUSSION THREAD

**Discussion by Eddiiie** [stm32duino/Arduino_Core_STM32#2209] [Variants] [answered]
Hi,
I spent the last 3 hours going through the WIki on how to add a variant ..  It looks like the board and the information was already present in the files I looked at, but the variant is not showing in Arduino IDE 2.

The chip is -  STM32L152VBT6

The board is an old Olimex STM32-P152

I see it in 
$\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.7.0\variants\STM32L1xx\L151V(8-B)(H-T)x(A)_L152V(8-B)(H-T)x(A)\boards_entry.txt

Before finding the there, I modified boards.txt

```
# Generic L152VBT6
GenL1.menu.pnum.GENERIC_L152VBTX=Generic L152VBTx
GenL1.menu.pnum.GENERIC_L152VBTX.upload.maximum_size=131072
GenL1.menu.pnum.GENERIC_L152VBTX.upload.maximum_data_size=32768
GenL1.menu.pnum.GENERIC_L152VBTX.build.board=GENERIC_L152VBT6
GenL1.menu.pnum.GENERIC_L152VBTX.build.product_line=STM32L152xB
GenL1.menu.pnum.GENERIC_L152VBTX.build.variant=STM32L1xx/L151V(8-B)(H-T)x(A)_L152V(8-B)(H-T)x(A)
```

Step 2.4 is not very clear..    the whole step was confusing.
`Copy the STM32YYxxxxxx_FLASH.ld generated by [STM32CubeMX](http://www.st.com/en/development-tools/stm32cubemx.html) in the variant folder and rename it: ldscript.ld`

I copied the FLASH.Id file to 
`$\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.7.0\variants\STM32L1xx\L151V(8-B)(H-T)x(A)_L152V(8-B)(H-T)x(A)
`
then renamed it to "ldscript.ld"

Is that correct?

Editing the ldscript.ld file blew my mind..  Having to generate the file, copy it, rename it, then edit it with more code ripped from platform.txt ?    Not sure I even got that right.

I think all of this is not necessary since it is already present in the boards file(s). ?    It just needs to show up in the "Board Part Number" menu?

I never got Blink to work!

**Comment by fpistm** [comment, marked as answer]
With Arduino IDE 2.x there is a bug, the menu is not refreshed and use the cached one.
See the workaround here:
https://github.com/arduino/arduino-ide/issues/1030#issuecomment-1152005617

Here an example on the way to add board:
https://github.com/stm32duino/Arduino_Core_STM32/pull/2042/commits/f0b57d65635c4a163582cb670c82a5852b5c1bce
